linux/LinuxProcessTable.c

```diff
@@ -795,12 +795,11 @@ static bool LinuxProcessTable_readSmapsFile(LinuxProcess* process, openat_arg_t
    while (fgets(buffer, sizeof(buffer), fp)) {
       if (!strchr(buffer, '\n')) {
          // Partial line, skip to end of this line
-         while (fgets(buffer, sizeof(buffer), fp)) {
-            if (strchr(buffer, '\n')) {
-               break;
-            }
+         if (!skipEndOfLine(fp)) {
+            fclose(fp);
```

Comment:
I suggest this error handling be moved to outside the loop, into its own `endOfFile:` label.

Reply:
As this is a single line, moving it to its own label just complicates things without really improving readability.